### DISCUSSION TITLE
Autocommand event when JDTLS started, analog to LspAttach

### DISCUSSION THREAD

**Discussion by bratekarate** [mfussenegger/nvim-jdtls#701] [Q&A]
I want to execute a command when JDTLS was started successfully. I can hook into `LspAttach` to execute commands after attach. However, JDTLS is not started yet at this point. Is there any way to "wait until JDTLS is started and then execute a command"?

**Comment by mfussenegger** [comment]
`jdtls` is started when `LspAttach` fires but loading the project/analysing the workspace might not be finished.

There's no standardized event in the language server protocol for that.
You could add a custom `handlers["language/status"]` handler to your `config` - it should eventually receive a `ServiceReady` message which you could react on

untested:

```lua
local config = {...} -- from readme

config.handlers["language/status"] = function(err, result)
  if result and result == "ServiceReady" then
    --- your logic
  end
end
```
